docs/src/components/hardware-list.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+
+import {
+  Board,
+  HardwareMetadata,
+  Interconnect,
+  Shield,
+} from "../hardware-metadata";
+
+interface HardwareListProps {
+  items: HardwareMetadata[];
+}
+
+function itemIds(item: HardwareMetadata) {
+  if (item.type == "board" || item.type == "shield") {
+    let nodes = (item.siblings || [item.id])
+      .map<JSX.Element | string>((id) => <code key={id}>{id}</code>)
+      .reduce(
+        (prev, curr, index) => [...prev, index > 0 ? " and " : "", curr],
+        [] as Array<string | JSX.Element>
+      );
+    return <span key={item.id}>{nodes}</span>;
+  } else {
+    return <code key={item.id}>{item.id}</code>;
+  }
+}
+
+function typeLabel(type: HardwareMetadata["type"]): string {
```

Comment:
I think you could simplify this to just a record like this:

```ts
// Type alias not necessary, but maybe easier to read
type HardwareType = HardwareMetadata['type'];

const TYPE_LABELS: Record<HardwareType, string> = {
  board: 'Board: ',
  shield: 'Shield: ',
  interconnect: 'Interconnect: ',
};
```

Reply:
Much nicer! Refactored to this approach.